File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/webapp/src/ds/ClassificationSystemLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.bd.standards.ds.model.ClassificationSystemLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"code", type:"string"},
		{name:"name", type:"string"},
		{name:"active", type:"boolean"},
		{name:"useInContext", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.bd.standards.ds.model.ClassificationSystemLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"code", type:"string"},
		{name:"name", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"useInContext", type:"string"}
	] 
});
	
Ext.define("net.nan21.dnet.module.bd.standards.ds.param.ClassificationSystemLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/standards/ds/model/ClassificationSystemLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.bd.standards.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeWithCodeLov;

import net.nan21.dnet.module.bd.standards.domain.entity.ClassificationSystem;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = ClassificationSystem.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = ClassificationSystemLovDs.fCODE) })
public class ClassificationSystemLovDs extends
        AbstractTypeWithCodeLov<ClassificationSystem> {

    public static final String fUSEINCONTEXT = "useInContext";

    @DsField(jpqlFilter = " :useInContext like e.useInContext ")
    private String useInContext;

    public ClassificationSystemLovDs() {
        super();
    }

    public ClassificationSystemLovDs(ClassificationSystem e) {
        super(e);
    }

    public String getUseInContext() {
        return this.useInContext;
    }

    public void setUseInContext(String useInContext) {
        this.useInContext = useInContext;
    }

}
